src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -271,12 +285,15 @@ internal async ValueTask<IEnumerable<Document>> GetAllRegularAndSourceGeneratedD
             var generatedDocumentState = generatedDocumentStates.GetState(documentId);
             if (generatedDocumentState != null)
             {
-                return ImmutableHashMapExtensions.GetOrAdd(ref _idToSourceGeneratedDocumentMap, generatedDocumentState.Id, s_createSourceGeneratedDocumentFunction, (generatedDocumentState, this));
+                return GetOrCreateSourceGeneratedDocument(generatedDocumentState);
             }
 
             return null;
         }
 
+        internal SourceGeneratedDocument GetOrCreateSourceGeneratedDocument(SourceGeneratedDocumentState state)
```

Comment:
Was this supposed to get called somewhere else too?

Reply:
It is: https://github.com/dotnet/roslyn/blob/af7a90f6404d8a34b8e58ede5210268ad41b6b31/src/Features/Core/Portable/EditAndContinue/EditSession.cs#L387